x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/QuerySettings.java

```diff
@@ -67,18 +127,20 @@ public static void validate(EsqlStatement statement, SettingsValidationContext c
                 throw new ParsingException(setting.source(), "Setting [" + setting.name() + "] is only available in snapshot builds");
             }
 
-            if (setting.value().dataType() != def.type()) {
+            if (def.type() != null && setting.value().dataType() != def.type()) {
                 throw new ParsingException(setting.source(), "Setting [" + setting.name() + "] must be of type " + def.type());
             }
 
-            Literal literal;
-            if (setting.value() instanceof Literal l) {
-                literal = l;
-            } else {
-                throw new ParsingException(setting.source(), "Setting [" + setting.name() + "] must have a literal value");
+            if (def.type() != null && setting.value().foldable() == false) {
```

Comment:
I would add a comment here explaining that this "type() != null" is because of MapParams not being foldable yet